src/test/ui/macros/macro-follow.rs

```diff
@@ -5,23 +5,23 @@
 
 // FOLLOW(pat) = {FatArrow, Comma, Eq, Or, Ident(if), Ident(in)}
 macro_rules! follow_pat {
-    ($p:pat ()) => {};       //~ERROR  `$p:pat` is followed by `(`
-    ($p:pat []) => {};       //~ERROR  `$p:pat` is followed by `[`
-    ($p:pat {}) => {};       //~ERROR  `$p:pat` is followed by `{`
-    ($p:pat :) => {};        //~ERROR `$p:pat` is followed by `:`
-    ($p:pat >) => {};        //~ERROR `$p:pat` is followed by `>`
-    ($p:pat +) => {};        //~ERROR `$p:pat` is followed by `+`
-    ($p:pat ident) => {};    //~ERROR `$p:pat` is followed by `ident`
-    ($p:pat $q:pat) => {};   //~ERROR `$p:pat` is followed by `$q:pat`
-    ($p:pat $e:expr) => {};  //~ERROR `$p:pat` is followed by `$e:expr`
-    ($p:pat $t:ty) => {};    //~ERROR `$p:pat` is followed by `$t:ty`
-    ($p:pat $s:stmt) => {};  //~ERROR `$p:pat` is followed by `$s:stmt`
-    ($p:pat $q:path) => {};  //~ERROR `$p:pat` is followed by `$q:path`
-    ($p:pat $b:block) => {}; //~ERROR `$p:pat` is followed by `$b:block`
-    ($p:pat $i:ident) => {}; //~ERROR `$p:pat` is followed by `$i:ident`
-    ($p:pat $t:tt) => {};    //~ERROR `$p:pat` is followed by `$t:tt`
-    ($p:pat $i:item) => {};  //~ERROR `$p:pat` is followed by `$i:item`
-    ($p:pat $m:meta) => {};  //~ERROR `$p:pat` is followed by `$m:meta`
+    ($p:pat ()) => {};       //~ERROR  `$p:pat2018` is followed by `(`
+    ($p:pat []) => {};       //~ERROR  `$p:pat2018` is followed by `[`
+    ($p:pat {}) => {};       //~ERROR  `$p:pat2018` is followed by `{`
+    ($p:pat :) => {};        //~ERROR `$p:pat2018` is followed by `:`
+    ($p:pat >) => {};        //~ERROR `$p:pat2018` is followed by `>`
+    ($p:pat +) => {};        //~ERROR `$p:pat2018` is followed by `+`
+    ($p:pat ident) => {};    //~ERROR `$p:pat2018` is followed by `ident`
+    ($p:pat $q:pat) => {};   //~ERROR `$p:pat2018` is followed by `$q:pat2018`
+    ($p:pat $e:expr) => {};  //~ERROR `$p:pat2018` is followed by `$e:expr`
+    ($p:pat $t:ty) => {};    //~ERROR `$p:pat2018` is followed by `$t:ty`
+    ($p:pat $s:stmt) => {};  //~ERROR `$p:pat2018` is followed by `$s:stmt`
+    ($p:pat $q:path) => {};  //~ERROR `$p:pat2018` is followed by `$q:path`
+    ($p:pat $b:block) => {}; //~ERROR `$p:pat2018` is followed by `$b:block`
+    ($p:pat $i:ident) => {}; //~ERROR `$p:pat2018` is followed by `$i:ident`
+    ($p:pat $t:tt) => {};    //~ERROR `$p:pat2018` is followed by `$t:tt`
+    ($p:pat $i:item) => {};  //~ERROR `$p:pat2018` is followed by `$i:item`
+    ($p:pat $m:meta) => {};  //~ERROR `$p:pat2018` is followed by `$m:meta`
```

Comment:
This looks like a regression in diagnostics to me.

Reply:
In general `pat2018` and `pat2021` look shameful enough to hide them as much as possible and not show to users unless really necessary.

Reply:
Hmm... ok, I was wondering what you would think of it. I'll change it.